lib/config/fileconf.go

```diff
@@ -437,6 +437,37 @@ func (conf *FileConfig) CheckAndSetDefaults() error {
 		}
 	}
 
+	matchers := make([]AWSMatcher, 0, len(conf.SSH.AWSMatchers))
+
+	for _, matcher := range conf.SSH.AWSMatchers {
+		if matcher.InstallParams == nil {
+			matcher.InstallParams = &InstallParams{
+				JoinParams: JoinParams{
+					TokenName: defaults.IAMInviteTokenName,
+					Method:    types.JoinMethodIAM,
+				},
+			}
+		} else {
+			method := matcher.InstallParams.JoinParams.Method
+			if method == "" {
+				matcher.InstallParams.JoinParams.Method = types.JoinMethodIAM
+			} else if method != types.JoinMethodIAM {
+				return trace.BadParameter("only IAM joining is supported for EC2 auto-discovery")
+			}
+			token := matcher.InstallParams.JoinParams.TokenName
+			if token == "" {
+				matcher.InstallParams.JoinParams.TokenName = defaults.IAMInviteTokenName
+			}
+		}
+
+		if matcher.SSM.Document == "" {
+			matcher.SSM.Document = defaults.AWSInstallerDocument
```

Comment:
doesn't SSM Doc need to be manually created for each AWS account? 
If that's the case should we force default values or should we just return an error either if empty or if the SSM doc does not exist

Same logic applies for IAM Token Name